recipes-kernel/linux/nilrt-nohz/README.nohz

```diff
@@ -0,0 +1,98 @@
+==============================
+NI Linux Real-Time NOHZ Kernel
+==============================
+
+This document describes the recommended application design, configuration
+options and boot parameters required for obtaining increased real-time (RT)
+performance when using the NILRT nohz kernel.
+
+The nohz kernel provided by 'packagegroup-ni-nohz-kernel' is a Linux kernel
+configured to omit scheduling-clock interrupts on CPUs that are either idle or
+that have only one runnable task (CONFIG_NO_HZ_FULL=y). This avoids the OS
+induced jitter caused by processing scheduling interrupts at the cost of
+increased user/kernel transitions among other things.
+
+Using this kernel configuration is advantageous for application designs where RT
+threads are assigned to dedicated CPU cores and spend most of their execution
+time in user-space running polling loops. Consult the full kernel
+documentation on this topic[1] for a list of pros/cons.
+
+
+Installation
+============
+
+Using a terminal console (serial, ssh):
+
+  * opkg update
+
+  * opkg packagegroup-ni-nohz-kernel
+
+
+Application Design Considerations
+=================================
+
+  * Use a single RT thread per reserved CPU.
+
+  * Use warm-up interrations. These are "dry-run" iterations where the
+    processing done by the RT threads is disregarded. This allows the kernel to
+    load all the data and code from disk into RAM and warm-up the CPU caches.
+
+  * Isolate the CPUs reserved for critical RT threads and mark them available
+    for adaptive-ticks by setting the 'isolcpus' and 'nohz_full' kernel boot
+    parameters.[2]
+    e.g.
+      fw_setenv othbootargs isolcpus=6,7 nohz_full=6,7
+
+    Note: On NI Linux Real-Time (NILRT) systems CPU 0 is configured to handle
+    general interrupts, kernel, and system tasks and should not be reserved for
+    critical RT tasks.
+
+  * Use CPU pools (also known as cgroup cpu sets) to configure CPUs reserved for
+    critical RT threads versus general system threads:
+
+      - in LabVIEW use the "Real-Time->RT SMP CPU Utilities" palette to
+        configure the CPU pools appropriately;
+
+      - when designing RT applications that are not LabVIEW based this can be
+        accomplished by setting the CPU mask of the 'system_set' and
+        'LabVIEW_tl_set'.
+	e.g.
```

Comment:
Do we use this format (introducing a code example with "e.g." on its own line and beginning the example on the next line) in other kernel readmes? If so, let it stand to be consistent.

Reply:
I am actually not sure what the standard format is. I am open to suggestions. I don't necessarily like the look of "e.g." on its own line either.

Reply:
Yeah, sorry, I didn't leave you with any suggestions, did I? :-)  I briefly glanced through some pages on the Linux Kernel site (https://www.kernel.org/doc/html/latest/), but their style seems to be dependent on HTML or Markdown -- code snippets are contained in a shaded box and use a different font. 

Obvs not a show-stopper to change these, but here are a couple suggestions, if you'd like some ideas:
- At the sentence preceding the example, add "For example:" or "Example follows," skip a line (as you did), and then add the code snippet.
- Replace "e.g." with "Example" or "Usage Example."